demo-apps/cf-android/app/src/main/java/org/eclipse/californium/examples/MainActivity.java

```diff
@@ -103,4 +123,19 @@ protected void onPostExecute(CoapResponse response) {
             }
         }
     }
+
+    private void initCoapEndpoint(String uri){
```

Comment:
Hm, this seems to be called for every request. Is this your intention?

Reply:
Javadoc for service, I can't see the relation to the UI thread or asyn task. Did you mix something up? And the rest of the comment is also a mystery for me.
In which thread can you call your initCoapEndpoint and in which thread not? 
And just check the exceptions! "p12" was also "not working", but in fact it works and the exception was caused by a different mistake. In my app an setup all endpoints in the EndpointManager, plain and secure, in a call based in onCreate(). So simply pass the uri to the coapclient and leave the logic, which endpoint is selected, to that. As I wrote now several times.